src/libraries/System.Private.Xml/src/System/Xml/XPath/XPathNavigator.cs

```diff
@@ -1566,7 +1566,7 @@ public virtual void DeleteSelf()
             DeleteRange(this);
         }
 
-        public virtual void PrependChildElement(string prefix, string localName, string namespaceURI, string value)
+        public virtual void PrependChildElement(string? prefix, string localName, string? namespaceURI, string value)
```

Comment:
value is being explicitly checked for null in all of these. Seems it should be nullable, too?

Reply:
Good spot. ReSharper normally highlights these for me, but I had 'solution wide analysis' off for this project. I turned it on and it's spotted several more. What is the best practice in PRs for changing related things? I understand nullable annotations are a rather special type of change, but when we're changing a method, should we look for related items in that method, or at the class level?  For instance, I can see `CheckValidityHelper` (ln 1082) needs nullable annotations on `_reader` and `_nextEventHandler`. SHould I change those too?

Reply:
Here's just some of the issues with nullability:

![image](https://user-images.githubusercontent.com/263416/178147747-53436735-d61e-46bd-ac05-86e5a7d0c606.png)

.. and an example from the the 1st entry:

![image](https://user-images.githubusercontent.com/263416/178147765-aaa00a76-956a-4b66-b9ff-eb9b8f162281.png)

Reply:
That whole block could be:

![image](https://user-images.githubusercontent.com/263416/178147921-bded2592-a302-4070-a8a7-b0fe32f9b8d8.png)

Anyway, I digress. I'm happy to go through this solution to tidy a few of them up, if that's something that the team thinks needs to be done.

Reply:
> and an example from the the 1st entry

That's a false positive: removing that check would break the implementation.

Right now I'm interested in fixing incorrect annotations on public surface area. Please feel free to do more of that from Xml in this PR. We can also do it separately. I was calling out the value case though because as long as we're fixing one parameter in a method, we should fix all of them.